src/test/java/StringManipulationTest.java

```diff
@@ -55,27 +63,32 @@ public void testRemoveNthCharacter2() {
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Bring");
+        assertEquals("Bri", manipulatedstring.removeNthCharacter(5, false));
```

Comment:
You are missing test cases to check for Exceptions being thrown. For example
 @Test //assert throws IllegalArgumentException if n == 0
    public void testRemoveNthCharacter3() {
        manipulatedstring.setString("123");
        assertThrows(IllegalArgumentException.class, () -> {manipulatedstring.removeNthCharacter(0,true);});
    }